extension/intents/tabs/tabs.js

```diff
@@ -104,6 +104,17 @@ intentRunner.registerIntent({
   },
 });
 
+intentRunner.registerIntent({
+  name: "tabs.closeAll",
+  async run(context) {
+    const activeTabs = await context.activeTabs();
+    for (const tab of activeTabs) {
+      await browser.tabs.remove(tab.id);
```

Comment:
You could actually remove all of them at once: `browser.tabs.remove(tabs.map(tab => tab.id))`

But closing all tabs wouldn't be right, it leaves Firefox in a weird state. I think it quits Firefox on Windows (maybe?) and on Mac it leaves Firefox open but with no windows.

Instead only tabs that aren't pinned should be closed, and only tabs in the current window. And maybe the active tab should also be left open. So that would be something like `browser.tabs.query({active: false, lastFocusedWindow: true, pinned: false})`

Reply:
done!